### PR TITLE
Update vt_scrape_function_author to more generic vt_scape_tags_from

### PR DESCRIPTION
**Pull request by thebioengineer**
Generalized  `vt_scrape_function_authors()` into `vt_scrape_tags_from()` to scrape authors and collect roxygen tags from any of the requirements, functions, test cases, and test code. Based on the `parse_roxygen()`/`vt_scrape_roxygen()` generics to handle both `@section last edited by`/`@section last edit date` and `@editor`/`@editDate`

### CODE REVIEW

File: R/scrape_authors.R [hunk 1]
```diff
@@ -20,20 +21,24 @@
...
+  ### - referencing source code
+  ### - running a validation report on a validated package 
+  if(is_package(pkg) | is_package(src) | Sys.getenv("vt_validation_state") == "installed" ){
     types <- c(types, "functions")
   }
 
   type <- match.arg(type,choices = types)
 
-  dir_ref <- if(type == "functions"){
+  ## Need this so that
```
Comment (mariev) [inline R/scrape_authors.R:37]:
```suggestion

```

remove extraneous comments

**Review by mariev — CHANGES_REQUESTED** [review]
It is not immediately clear when a user should be using pkg vs src argument to point to files. I see in the code that each feeds slightly differently into `dir_ref` argument, but wouldn't the logic hold with a single location input? 

Is it possible to scrape everything at once if a vector is passed - maybe we can write a wrapper to handle this?

getting this error when running for `type == "functions"`

```
> basename(getwd())
[1] "valtools"
> vt_scrape_tags_from(type = "functions")
Error in .f(.x[[i]], ...) : 
  R6 class (vt_dynamic_referencer) without source references. If you use the `installed` load method in `DESCRIPTION`, then try re-installing the package with option '--with-keep.source'. E.g. `install.packages(..., INSTALL_OPTS = "--with-keep.source")`.
> packageDescription("valtools")$GithubRef
[1] "scrape_authors"
```

when `type == "test_code"` requires {testthat} to be loaded.

```
> vt_scrape_tags_from(type = "test_code")
Error in eval_bare(node_car(expr), env) : object 'test_that' not found

> library(testthat)
Warning message:
package ‘testthat’ was built under R version 4.0.3 
> vt_scrape_tags_from(type = "test_code")
[[1]]
   test_code    editor   editDate
1 TESTNUMBER An editor 2021-03-09

```

File: R/scrape_authors.R (hunk 1 above)
Comment (thebioengineer) [reply R/scrape_authors.R:37]: I completed the comment

**Comment by thebioengineer** [comment]
The first error and second errors are related, and I have resolved. It was trying to parse and evaluate the code itself rather than getting just the roxygen tags. Which is why for the second example we got an error when testthat was not loaded.

Obviously I don't want it to evaluate, so I have stopped that by setting the env to NULL in parse_text() and have now applied it everywhere. 

We might want to add the disclaimer that for R6 objects support is limited to a single editor.

**Comment by mariev** [comment]
> We might want to add the disclaimer that for R6 objects support is limited to a single editor.

I am able to add multiple editors using comma separation. E.g.

```
withr::with_tempdir({

  vt_create_package( open = FALSE, username_list =list(vt_user(
    name = "test",
    title = "test",
    role = "tester",
    username = "mvendett"
  )) )
  vt_use_test_code("test_code1")

  this_test_code <- readLines(file.path(getwd(), "vignettes", "validation", "test_code", "test_code1.R"))
  this_test_code[5] <- "#' @editor one_author, two_author"
  writeLines(this_test_code,
             file.path(getwd(), "vignettes", "validation", "test_code", "test_code1.R") )
  tags <- vt_scrape_tags_from(type = "test_code", ref = "vignettes/validation")
  tags
})

...
[[1]]
   test_code                 editor   editDate
1 TESTNUMBER one_author, two_author 2021-03-10
```

**Comment by thebioengineer** [comment]
yeah, through commas we can definitely add more. I was thinking about how roxygen now supports R6 documentation at the individual method level now, which if the person thought that then they could pass an author for each method, we don't support it right now

**Comment by mariev** [comment]
👍 definitely. We should be setting the expectation that `@editor` etc are treated more like `@title` vs `@param`

**Comment by mariev** [comment]
@thebioengineer added some minor updates to documentation. 👍  or 👎 ?

**Comment by thebioengineer** [comment]
> @thebioengineer added some minor updates to documentation. 👍 or 👎 ?

Looks good to me. I think that is helpful to capture!

**Review by mariev — APPROVED** [review]

**Pull request merged**